File: 02_algorithm/programmers/Level2/피보나치 수.py
def solution(n):
    arr = [0] * (n + 1)
    arr[0], arr[1] = 0, 1
    if n <= 1:
        return arr[n]
    for i in range(2, n + 1):
        arr[i] = (arr[i - 1] + arr[i - 2]) % 1234567
    return arr[n] % 1234567

print(solution(5))

File: 06_js/04_algorithm/programmers/Level2/피보나치 수.js
function solution(n) {
  let arr = Array(n + 1).fill().map(_ => 0);
  arr[0] = 0;
  arr[1] = 1;
  
  if (n <= 1) {
    return arr[n];
  }
  
  for (let i = 2; i <= n; i++) {
    arr[i] = (arr[i - 1] + arr[i - 2]) % 1234567;
  }
  
  return arr[n] % 1234567;
}

console.log(solution(3)); // 2
console.log(solution(5)); // 5